### ISSUE TITLE
Update TOC to match Flutter doc's

### ISSUE THREAD

**Issue by parlough** [dart-lang/site-www#6838]
The Flutter doc site's TOC was overhauled with a new design with improved accessibility, a new style, and a mobile version. After the Jaspr migration, Dart's TOC should be updated to match or be similar.

Flutter reference: https://github.com/flutter/website/pull/11701
